l10n_es_aeat_sii/models/account_invoice.py

```diff
@@ -954,8 +966,23 @@ def _connect_sii(self, mapping_key):
         session.cert = (public_crt, private_key)
         transport = Transport(session=session)
         history = HistoryPlugin()
-        client = Client(wsdl=wsdl, transport=transport, plugins=[history])
-        return client.bind('siiService', port_name)
+        client = Client(
+            wsdl=params['wsdl'], transport=transport, plugins=[history],
+        )
+        return self._bind_sii(
+            client, params['port_name'], params['test_address'],
+        )
+
+    @api.multi
+    def _bind_sii(self, client, port_name, test_address):
+        self.ensure_one()
+        if test_address:
+            service = client._get_service('siiService')
```

Comment:
Por qué cambiar la manera en la que se crea el cliente para las pruebas?

Reply:
En las haciendas forales no están definidos los entornos de pruebas a nivel de wsdl. Por eso lo he hecho así

wsdl estatal
![image](https://user-images.githubusercontent.com/14833643/40381965-69628500-5dfd-11e8-8ed0-424b35aff886.png)

wsdl gipuzkoa
![image](https://user-images.githubusercontent.com/14833643/40382019-95db6c8c-5dfd-11e8-9052-028c9ede6a50.png)

Reply:
Este enlace está relacionado con las soluciones que vi en su momento:
https://stackoverflow.com/questions/42236251/change-service-url-in-python-zeep

Al final es lo que estoy usando, pero primero llamo al método _get_port para saber el nombre completo.
Ya te digo que no encontré nada mejor. Igual @acysos nos puede ayudar con esto

Reply:
Bueno, si tal vez no merece la pena entrar tanto en ello. Sí que quería evitar este método extra, pero bueno, vistas las razones, podemos dejarlo así.

Reply:
Si es por el nuevo método, se puede dejar en el mismo método todo junto

Reply:
Me refería al código extra que implica el nuevo método. Y no se pueden poner esas 3 líneas igual para el entorno de producción? Porque entonces la solución mejor es poner directamente eso. La idea es evitar bifurcaciones de si es test o no en esta fase.

Reply:
Sería algo así:
```
    @api.multi
    def _bind_sii(self, client, port_name, test_address=None):
        self.ensure_one()
        service = client._get_service('siiService')
        port = client._get_port(service, port_name)
        address = test_address or port.binding_options['address']
        return client.create_service(port.binding.name, address)
```
¿Como lo ves?

Reply:
He hecho cambios según el comentario anterior